src/Conversion/InterventionImageFileConverter.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace SilverStripe\Assets\Conversion;
+
+use Intervention\Image\Exception\ImageException;
+use SilverStripe\Assets\Image_Backend;
+use SilverStripe\Assets\InterventionBackend;
+use SilverStripe\Assets\Storage\AssetStore;
+use SilverStripe\Assets\Storage\DBFile;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * File converter powered by the Intervention Image library.
+ * Supports any file conversions that Intervention Image can perform.
+ */
+class InterventionImageFileConverter implements FileConverter
+{
+    public function supportsConversion(string $fromFormat, string $toFormat, array $options = []): bool
+    {
+        $unsupportedOptions = $this->validateOptions($options);
+        if (!empty($unsupportedOptions)) {
+            return false;
+        }
+        // This converter requires intervention image as the image backend
+        $backend = Injector::inst()->get(Image_Backend::class);
+        if (!is_a($backend, InterventionBackend::class)) {
+            return false;
+        }
+        return $this->supportedByIntervention($fromFormat, $backend) && $this->supportedByIntervention($toFormat, $backend);
+    }
+
+    public function convert(DBFile $from, string $toFormat, array $options = []): DBFile
+    {
+        // Do some basic validation up front for things we know aren't supported
+        $problems = $this->validateOptions($options);
```

Comment:
Above we call the variable `unsupportedOptions` which sounds a bit better, however it's not clear that a method called `validateSomething` would return a list of exclusions/unsupported items only. Could we consider something like `filterInvalidOptions` or return an array of valid and invalid options etc.

Reply:
`filterInvalidOptions` suggests it will filter out anything that's invalid and return the valid options (if there were any) as a result. I have no problem renaming the method or the variable but I think that name suggests different functionality.

The idea here is that there's a clear exception if any unsupported, invalid, etc options are passed in so developers can set their options correctly. No unsupported/invalid options will ever be passed in here unless the developer explicitly chooses to use this converter directly - the manager would have checked `supportsValidation()` first, which would have returned `false`.

> however it's not clear that a method called validateSomething would return a list of exclusions/unsupported items only.

It's returning an array of problems (or exception messages if you prefer?) so we can tell the developer exactly what they did wrong. I'm not sure what method name could convey that. `checkIfOptionsAreValidAndReturnClearMessagesAboutThingsTheDeveloperNeedsToFix()` is a bit too verbose, for example.

It's a private method anyway so it's not as important as a public method would be, and IMO it's pretty clear from its usage what it's doing - I can add a PHPdoc if it really needs clarifying, or if somsone can think of a really good name I'm happy to change it.

Reply:
Fair enough re: all the points, thanks for taking the time to explain that!